### PR TITLE
EPMRPP-86794 || OAuth test connection fails when opening integrations…

### PR DESCRIPTION
**Pull request by BlazarQSO**
… page

### CODE REVIEW

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx [hunk 1]
```diff
@@ -62,6 +63,18 @@ export const IntegrationSettings = (props) => {
...
+    data,
+    onUpdate,
+    formFieldsComponent,
+    editAuthConfig,
+    isEmptyConfiguration,
+    formKey,
+    isGlobal,
+  } = props;
+
+  const pluginName = props.data.integrationType?.name;
```
Comment (Bam6ycha) [inline app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx:76]: Looks like it is better to extract plugin name from data using destructuring.
 const {
    data,
    onUpdate,
    formFieldsComponent,
    editAuthConfig,
    isEmptyConfiguration,
    formKey,
    isGlobal,
    data:{integrationType:{name:pliginName}}={}
  } = props;

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx [hunk 2]
```diff
@@ -74,8 +87,7 @@ export const IntegrationSettings = (props) => {
   );
 
   const testIntegrationConnection = useCallback(() => {
-    setLoading(true);
-    if ('id' in props.data) {
+    if ('id' in props.data && pluginName !== AD && pluginName !== LDAP && pluginName !== SAML) {
```
Comment (Bam6ycha) [inline app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx:90]: As for me, such a long condition looks bad.

May be in is better f.e. create an object such as authPlugins={
[AD]:AD,
...
}
and use it like this 
if('id' in data && !authPlugins[pluginName])

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx [hunk 3]
```diff
@@ -62,6 +63,18 @@ export const IntegrationSettings = (props) => {
   const availableProjectIntegrations = projectIntegrations[query.subPage] || [];
   const groupedIntegrations = [...availableGlobalIntegrations, ...availableProjectIntegrations];
```
Comment (Bam6ycha) [inline app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx:64]: Could you fix this SonarQube warning?

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

**Review by AmsterGet — CHANGES_REQUESTED** [review]
Please check the integration Settings components by the next path: src/components/integrations/integrationProviders.
Inside of the `SamlSettings`, `LdapSettings`, `ActiveDirectorySettings` the `preventTestConnection` prop exists.
So I suggest to just implement this prop usage inside of the `IntegrationSettings` to not hardcode the names of integrations that don't need to check connections, but hold this responsibility on each particular plugin implementation.

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx (hunk 1 above)
Comment (BlazarQSO) [reply app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx]: Reverted all changes

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx (hunk 2 above)
Comment (BlazarQSO) [reply app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx]: Reverted all changes

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx (hunk 3 above)
Comment (BlazarQSO) [reply app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx]: Reverted all changes

Comment (BlazarQSO) [reply app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx]: Fixed warning

File: app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx [hunk 4]
```diff
@@ -58,9 +58,18 @@ export const IntegrationSettings = (props) => {
   const query = useSelector(querySelector);
   const dispatch = useDispatch();
   const { trackEvent } = useTracking();
-  const availableGlobalIntegrations = globalIntegrations[query.subPage] || [];
-  const availableProjectIntegrations = projectIntegrations[query.subPage] || [];
-  const groupedIntegrations = [...availableGlobalIntegrations, ...availableProjectIntegrations];
+  const availableGlobalIntegrations = useMemo(() => globalIntegrations[query.subPage] || [], [
```
Comment (Bam6ycha) [inline app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx:61]: Could we combine these 3 useMemo into one?

Comment (BlazarQSO) [reply app/src/components/integrations/elements/integrationSettings/integrationSettings.jsx]: Done.

**Review by Bam6ycha — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**